daemon/entrypoint.sh

```diff
@@ -420,6 +424,102 @@ function osd_activate {
   fi
 }
 
+#############
+# OSD_DISKS #
+#############
+function get_osd_dev {
+  for i in ${OSD_DISKS}
+   do
+    osd_id=$(echo ${i}|sed 's/\(.*\):\(.*\)/\1/')
+    osd_dev="/dev/$(echo ${i}|sed 's/\(.*\):\(.*\)/\2/')"
+    if [ ${osd_id} = ${1} ]; then
+      echo -n "${osd_dev}"
+    fi
+  done
+}
+
+function set_osd_run {
+  mkdir -p /etc/service/${1}-${2}
+  cat >/etc/service/${1}-${2}/run <<EOF
+#!/bin/bash
+echo "store-daemon: starting daemon on ${HOSTNAME}..."
+exec /usr/bin/ceph-osd ${CEPH_OPTS} -f -d -i ${2} --setuser ceph --setgroup disk
+EOF
+  chmod +x /etc/service/${1}-${2}/run
+}
+
+function osd_disks {
+  if [[ ! -d /var/lib/ceph/osd ]]; then
+    echo "ERROR- could not find the osd directory, did you bind mount the OSD data directory?"
+    echo "ERROR- use -v <host_osd_data_dir>:/var/lib/ceph/osd"
+    exit 1
+  fi
+
+  if [[  -z ${OSD_DISKS} ]]; then
+    echo "ERROR- could not find the osd devices, did you configure OSD disks?"
+    echo "ERROR- use -e OSD_DISKS=\"0:sdd 1:sde 2:sdf\""
+    exit 1
+  fi
+
+  # make sure ceph owns the directory
+  chown ceph. /var/lib/ceph/osd
+
+  # check if anything is there, if not create an osd with directory
+  if [[ -z "$(find /var/lib/ceph/osd -prune -empty)" ]]; then
+    echo "Mount existing and prepared OSD disks for ceph-cluster ${CLUSTER}"
+    for OSD_ID in $(ls /var/lib/ceph/osd |  awk 'BEGIN { FS = "-" } ; { print $2 }'); do
+      OSD_DEV=$(get_osd_dev ${OSD_ID})
+      if [[ -z ${OSD_DEV} ]]; then
+        echo "No device mapping for ${CLUSTER}-${OSD_ID} for ceph-cluster ${CLUSTER}"
+        exit 1
+      fi
+      mount ${MOUNT_OPTS} $(dev_part ${OSD_DEV} 1) /var/lib/ceph/osd/${CLUSTER}-${OSD_ID}/
+      xOSD_ID=$(cat /var/lib/ceph/osd/${CLUSTER}-${OSD_ID}/whoami)
+      if [[ "${OSD_ID}" != "${xOSD_ID}" ]]; then
+        echo "Device ${OSD_DEV} is corrupt for /var/lib/ceph/osd/${CLUSTER}-${OSD_ID}"
+        exit 1
+      fi
+      set_osd_run ${CLUSTER} ${OSD_ID}
+    done
+    exec /sbin/my_init
+  else
+    for i in ${OSD_DISKS}; do
+      OSD_ID=$(echo ${i}|sed 's/\(.*\):\(.*\)/\1/')
+      OSD_DEV="/dev/$(echo ${i}|sed 's/\(.*\):\(.*\)/\2/')"
+      if [[ "$(parted --script ${OSD_DEV} print | egrep '^ 1.*ceph data')" && ${OSD_FORCE_ZAP} -ne "1" ]]; then
+        echo "ERROR- It looks like this device is an OSD, set OSD_FORCE_ZAP=1 to use this device anyway and zap its content"
+        exit 1
+      elif [[ "$(parted --script ${OSD_DEV} print | egrep '^ 1.*ceph data')" && ${OSD_FORCE_ZAP} -eq "1" ]]; then
+        ceph-disk -v zap ${OSD_DEV}
+      fi
+
+      if [[ ! -z "${OSD_JOURNAL}" ]]; then
+        ceph-disk -v prepare ${OSD_DEV} ${OSD_JOURNAL}
```

Comment:
This is for new feature (I've called it DISK_TYPE=devices), when needed zap&prepared&activate). In environment need add variable like OSD_DISKS="0:sda 1:sdb 2:sdc"